slice/internal/controller/workload_controller.go

```diff
@@ -110,18 +110,20 @@ func (r *WorkloadReconciler) newSlice(wl *kueue.Workload) (*v1alpha1.Slice, erro
 
 	if wl.Status.Admission != nil && wl.Status.Admission.PodSetAssignments != nil {
 		for _, psa := range wl.Status.Admission.PodSetAssignments {
-			for _, domain := range psa.TopologyAssignment.Domains {
-				if slice.Spec.NodeSelector == nil {
-					slice.Spec.NodeSelector = make(map[string][]string)
-				}
-				if slice.Spec.NodeSelector[TPUReservationSubblockLabel] == nil {
-					slice.Spec.NodeSelector[TPUReservationSubblockLabel] = []string{}
-				}
-				// make sure there are no duplicates in the nodeSelector
-				for _, v := range domain.Values {
-					exists := slices.Contains(slice.Spec.NodeSelector[TPUReservationSubblockLabel], v)
-					if !exists {
-						slice.Spec.NodeSelector[TPUReservationSubblockLabel] = append(slice.Spec.NodeSelector[TPUReservationSubblockLabel], v)
+			if psa.TopologyAssignment != nil {
```

Comment:
It can be possible that Workload have PodSetAssignments but doesn't have TopologyAssignment.